src/Core/Content/Product/Subscriber/ProductSubscriber.php

```
@@ -158,6 +164,71 @@ public function beforeWriteProduct(EntityWriteEvent $event): void
         }
     }
 
+    public function beforeDeleteProduct(EntityDeleteEvent $event): void
```

as you only use the connection directly, we should also write an integration test for the subscriber, to actually ensuring that it is working correctly. 

right now, the unit test depends highly on the mock data you give and this could be error prone